### PR TITLE
Prac 06 feedback

### PR DESCRIPTION
**Pull request by XanderCaubat**
Hello @trijeetbhandula,

Could you please review my code?

Thank you in advance :)

### CODE REVIEW

File: prac_06/car.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+"""
+
+
+class Car:
+    """Represent a Car object."""
+
+    def __init__(self, name="", fuel=0):
+        """Initialise a Car instance.
+
+        fuel: float, one unit of fuel drives one kilometre
```
Comment (trijeetbhandula) [inline prac_06/car.py:16]: It can be added as a comment except than being a part of docstring

File: prac_06/car.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        """
+        if distance > self.fuel:
+            distance = self.fuel
+            self.fuel = 0
+        else:
+            self.fuel -= distance
+        self._odometer += distance
+        return distance
+
+    def display(self):
```
Comment (trijeetbhandula) [inline prac_06/car.py:40]: Generally we can call it as __str__ method

File: prac_06/programming_language.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+"""
+CP1404/CP5632 - Practical 5
```
Comment (trijeetbhandula) [inline prac_06/programming_language.py:2]: Its practical 6 buddy 😄

File: prac_06/programming_language.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+Programming Languages
+Class for the programming language
+Estimate: 100 minutes
+Actual:    30 minutes
+"""
+
+
+class ProgrammingLanguage:
```
Comment (trijeetbhandula) [inline prac_06/programming_language.py:10]: Can add class docstring, not necessary.

File: prac_06/languages.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ruby = ProgrammingLanguage("Ruby", "Dynamic", True, 1995)
+visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+print(python)
+
+
+languages = [python, ruby, visual_basic]
+print("The dynamically languages are:")
+for language in languages:
+    if language.is_dynamic():
+        print(language.name)
```
Comment (trijeetbhandula) [inline prac_06/languages.py:21]: Excellent work 👍

File: prac_06/guitar.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+"""
+CP1404/CP5632 - Practical 6
+Guitar
+This program uses Guitar class as an example
+Estimate: 100 minutes
+Actual:    25 minutes
+"""
+from datetime import datetime
```
Comment (trijeetbhandula) [inline prac_06/guitar.py:8]: Great use, excellent idea to use this 🥇

File: prac_06/guitar.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+VINTAGE_AGE = 50
+current_date_time = datetime.now()
+
+
+class Guitar:
+
+    def __init__(self, name="", year_model=0, cost=0):
+        self.name = name
+        self.year_model = year_model
```
Comment (trijeetbhandula) [inline prac_06/guitar.py:18]: Good naming

File: prac_06/guitar.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        return f"{self.name} ({self.year_model}) : ${self.cost}"
+
+    def get_age(self):
+        return current_date_time.year - self.year_model
+
+    def is_vintage(self):
+        return self.get_age() >= VINTAGE_AGE
+
+    def __lt__(self, other):
+        return self.year_model < other.year_model
```
Comment (trijeetbhandula) [inline prac_06/guitar.py:31]: Where are we using this??

File: prac_06/guitar_test.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
+from prac_06.guitar import Guitar
+
+
+gibson = Guitar("Gibson L-5 CES", 1922, 16035.40)
+another_guitar = Guitar("Another Guitar", 2013, 700.00)
+print(f"{gibson.name} get_age() - Expected 100. Got {gibson.get_age()}")
+print(f"{another_guitar.name} get_age() - Expected 9. Got {another_guitar.get_age()}")
+print(f"{gibson.name} is_vintage() - Expected True. Got {gibson.is_vintage()}")
+print(f"{another_guitar.name} is_vintage() - Expected False. Got {another_guitar.is_vintage()}")
```
Comment (trijeetbhandula) [inline prac_06/guitar_test.py:9]: Great job 💯

File: prac_06/guitars.py [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+def get_year():
+    is_valid = False
+    while not is_valid:
+        try:
+            year = int(input("Year: "))
+            is_valid = True
+        except ValueError:
+            print("Invalid integer!")
+    return year
```
Comment (trijeetbhandula) [inline prac_06/guitars.py:46]: Extra marks for handling exceptions 😄

File: prac_06/guitars.py [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def display_guitars(guitars):
+    print("These are my guitars:")
+    guitars.sort()
+    for i, guitar in enumerate(guitars, 1):
+        vintage_string = ""
+        if guitar.is_vintage():
+            vintage_string = "(vintage)"
+        print(f"Guitar {i}: {guitar.name:>20} ({guitar.year_model}), worth ${guitar.cost:10,.2f}{vintage_string}")
```
Comment (trijeetbhandula) [inline prac_06/guitars.py:61]: Will this display in the way as it was expected in the sample output?
Like this:- 
Name:

... snip ...

These are my guitars:
Guitar 1:       Gibson L-5 CES (1922), worth $ 16,035.40 (vintage)
Guitar 2:        Line 6 JTV-59 (2010), worth $  1,512.90
Guitar 3:  Fender Stratocaster (2014), worth $    765.40

**Review by trijeetbhandula — COMMENTED** [review]
Great work 👍 Learnt a lot from your code.
Just maybe needs small changes 😄

File: prac_06/car.py (hunk 1 above)
Comment (XanderCaubat) [reply prac_06/car.py:16]: Cheers, will keep in mind for that one XD

File: prac_06/programming_language.py (hunk 3 above)
Comment (XanderCaubat) [reply prac_06/programming_language.py:2]: Was not reviewing properly, will check my next prac to make sure its the right prac

File: prac_06/languages.py (hunk 5 above)
Comment (XanderCaubat) [reply prac_06/languages.py:21]: Thank you!

**Pull request merged**